### DISCUSSION TITLE
How to find co-ordinates for My Location.

### DISCUSSION THREAD

**Discussion by blackadderkate** [blackadderkate/weather-widget-2#64] [General]
If you cannot find a location in my database but you _could_ find it using yr.no, please try the following:
1) [Download the database from Yr.No](https://www.yr.no/storage/lookup/English.csv.zip).
2) Extract the files inside into a suitable folder.
3) Open the file that matches the country you need to search (e.g. ES_spain.csv)
3) Search for your location in the first column.
4) Select and _copy_ the second column to the clipboard (e.g. lat=38.5345&lon=-0.13312&altitude=1).
5) In the widget, select Configure, then the Add a new Met.No location button, and _paste_ the second column data into the URL box.
6) Give the location a name, set the Timezone (e.g. Europe/Madrid), and you're done!

In case you are wondering why I didn't use this database in my widget, it is 552Mb when decompressed, and searching for "New York" returns 15673 matches which is just way too many to be useful - and I couldn't figure out an algorithm to decide which were important entries, and which were unnecessary...

**Comment by solbu** [comment]
I would disagree with the view that 15673 matches for New York is to many. Remember that «New York» is not just a city, it is also a US State, with lots of cities, small towns and other places.

Remember that the page that linked to the huge database said that it contains every place that was actively being pulled from their API in the last year only. I just did a check, and there are 26 places named «New York» in 16 states other than the state of New York. So there are 15673 different people that for some reason used YR to fetch their weather data.

**Reply by blackadderkate** [reply]
15673 entries is definitely too many results to return to an end-user when they do a search!

Basically I just wanted to try to explain to end-users _without getting technical_ why I couldn't just include the Yr.No database in the widget.

I know New York is also a state, which is why I deliberately used it as my example - New York is probably the worst place you could try to search for using the Yr.No CSV Database files, yet it is also a location that many people _might_ search for using this widget.

**Reply by solbu** [reply]
I live in Steinkjer municipality, but not in the city. My house is 12 miles away from the city center. So I would search for my local place, which is in the met.no database.
In the MET database there is 12661 locations withing my municipality alone, which has 24152 inhabitants according to last years census.

My point is that those who live in the state of New York don't search for New York, but their town or local area, and those in New York City don't search for plain New York, they do a narrow search for their location within the city.

Perhaps we could put the database in a separate repo, and maybe make it searchable.
And in the long run maybe implement that online database in the widget search feature.
The widget only works when connected to the internet anyway.

Just throwing an idea in the air, and may it fall where it will. :-)

**Comment by blackadderkate** [comment]
As a sports fan, I will often add generic locations where Formula One/Indycar races and Cricket/Tennis tournaments are taking place - including New York for the US Open tennis. A generic location is usually good enough to tell me if there's likely to be any rain or not, but to be fair that's probably not a common use for this widget!

I'm pretty happy with the current Database to be honest. It is based on the Cities500.zip file from GeoNames.org, and contains every location with a population > 500 people. I think this will satisfy the majority of our users, whilst still being small enough to bundle with the widget...

For users who want to add smaller locations not listed in the database, perhaps some notes on how to find their Latitude & Longitude with tools such as Google Maps or [Latitude and Longitude Finder](https://www.latlong.net/) would be useful. I'll add it to the list...
